src/Agent.Worker/Build/SvnCommandManager.cs

```diff
@@ -0,0 +1,828 @@
+
+        /// <summary>
+        /// Finds a local path the provided server path is mapped to.
+        /// </summary>
+        /// <param name="serverPath"></param>
+        /// <param name="rootPath"></param>
+        /// <returns></returns>
+        string ResolveServerPath(string serverPath, string rootPath);
+    }
+
+    public class SvnCommandManager : AgentService, ISvnCommandManager
+    {
+        public void Init(
+            IExecutionContext context,
+            ServiceEndpoint endpoint,
+            CancellationToken cancellationToken)
+        {
+            // Validation.
+            ArgUtil.NotNull(context, nameof(context));
+            ArgUtil.NotNull(endpoint, nameof(endpoint));
+            ArgUtil.NotNull(cancellationToken, nameof(cancellationToken));
+
+            ArgUtil.NotNull(endpoint.Url, nameof(endpoint.Url));
+            ArgUtil.Equal(true, endpoint.Url.IsAbsoluteUri, nameof(endpoint.Url.IsAbsoluteUri));
+            ArgUtil.NotNull(endpoint.Data, nameof(endpoint.Data));
+
+            ArgUtil.NotNull(endpoint.Authorization, nameof(endpoint.Authorization));
+            ArgUtil.NotNull(endpoint.Authorization.Parameters, nameof(endpoint.Authorization.Parameters));
+            ArgUtil.Equal(EndpointAuthorizationSchemes.UsernamePassword, endpoint.Authorization.Scheme, nameof(endpoint.Authorization.Scheme));
+
+            _context = context;
+            _endpoint = endpoint;
+            _cancellationToken = cancellationToken;
+
+            //find git in %Path%
+            IWhichUtil whichTool = HostContext.GetService<IWhichUtil>();
+            string svnPath = whichTool.Which("svn");
+
+            if (string.IsNullOrEmpty(svnPath))
+            {
+                throw new Exception(StringUtil.Loc("SvnNotInstalled"));
+            }
+            else
+            {
+                _context.Debug($"Found svn installation path: {svnPath}.");
+                _svn = svnPath;
+            }
+
+            // External providers may need basic auth or tokens
+            endpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.Username, out _username);
+            endpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.Password, out _password);
+
+            _acceptUntrusted = endpoint.Data.ContainsKey(/* WellKnownEndpointData.SvnAcceptUntrustedCertificates */ "acceptUntrustedCerts") &&
+            StringUtil.ConvertToBoolean(endpoint.Data[/* WellKnownEndpointData.SvnAcceptUntrustedCertificates */ "acceptUntrustedCerts"], defaultValue: false);
+        }
+
+        public async Task<string> UpdateWorkspace(
+            string rootPath,
+            Dictionary<string, SvnMappingDetails> distinctMappings,
+            bool cleanRepository,
+            string sourceBranch,
+            string revision)
+        {
+            if (cleanRepository)
+            {
+                RecreateDirectory(rootPath);
```

Comment:
Just something to consider: You may want to add diag tracing here or possibly even an output message to the build log. Deleting large directories may take a while, and it would help the end user understand what the build is doing. If feels like this step is more than just an internal implementation detail. It feels like it's an important detail from the end user perspective.